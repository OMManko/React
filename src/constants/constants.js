export const endPoint = 'http://react-cdp-api.herokuapp.com/movies';

export const moviePosterSizes = {
    SMALL: 'moviePoster__small',
    MEDIUM: 'moviePoster__medium'
};

export const sortOptions = {
    RELEASE: 'release',
    RATING: 'vote_average'
};

export const filterOptions = {
    TITLE: 'title',
    GENRE: 'genres'
};


export const actionTypes = {
    SEARCH_BY: 'SEARCH_BY',
    SORT_BY: 'SORT_BY',
    SEARCH_MOVIES: 'SEARCH_MOVIES',
    UPDATE_INPUT_VALUE: 'UPDATE_INPUT_VALUE',
    SET_SELECTED_MOVIE: 'SET_SELECTED_MOVIE',
    RESET_SELECTED_MOVIE: 'RESET_SELECTED_MOVIE'
};

export const initialState = {
    moviesList: [],
    moviesFound: 0,
    searchInputValue: '',
    selectedSortOption: sortOptions.RELEASE,
    selectedFilterOption: filterOptions.TITLE,
    selectedMovieInfo: {}
};
