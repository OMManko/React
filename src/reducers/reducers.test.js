import React from 'react';
import movieAppReducer from './reducers';
import { actionTypes, filterOptions, sortOptions } from "../constants/constants";
import { initialState } from "../constants/constants";

describe('<movieAppReducer />', () => {
    it('should return the initial state', () => {
        expect(movieAppReducer(undefined, {})).toMatchSnapshot();
    });

    it('should handle SEARCH_MOVIES', () => {
         const results = {
             data: [{ id: 123, title: 'Some movie' }, { id: 256, title: 'Some movie 2' }],
             total: 12
         };
         const action = { type: actionTypes.SEARCH_MOVIES, results };
         const expectedState = { ...initialState, moviesList: results.data, moviesFound: results.total };
         expect(movieAppReducer(initialState, action)).toEqual(expectedState);
    });
    it('should handle SEARCH_BY', () => {
        const action = { type: actionTypes.SEARCH_BY, option: filterOptions.GENRE };
        const expectedState = { ...initialState, selectedFilterOption: filterOptions.GENRE };
        expect(movieAppReducer(initialState, action)).toEqual(expectedState);
    });

    it('should handle SORT_BY', () => {
        const action = { type: actionTypes.SORT_BY, option: sortOptions.RATING };
        const expectedState = { ...initialState, selectedSortOption: sortOptions.RATING };
        expect(movieAppReducer(initialState, action)).toEqual(expectedState);
    });

    it('should handle UPDATE_INPUT_VALUE', () => {
        const input = 'test';
        const action = { type: actionTypes.UPDATE_INPUT_VALUE, value: input };
        const expectedState = { ...initialState, searchInputValue: input };
        expect(movieAppReducer(initialState, action)).toEqual(expectedState);
    });

    it('should handle SET_SELECTED_MOVIE', () => {
        const selectedMovie = { id: 123, title: 'Test' };
        const action = { type: actionTypes.SET_SELECTED_MOVIE, movie: selectedMovie };
        const expectedState = { ...initialState, selectedMovieInfo: selectedMovie };
        expect(movieAppReducer(initialState, action)).toEqual(expectedState);
    });

    it('should handle RESET_SEARCH', () => {
        const selectedMovie = { id: 123, title: 'Test' };
        const action = { type: actionTypes.RESET_SEARCH, selectedMovieInfo: selectedMovie };
        const expectedState = { ...initialState, selectedMovieInfo: {} };
        expect(movieAppReducer(initialState, action)).toEqual(expectedState);
    });
});
