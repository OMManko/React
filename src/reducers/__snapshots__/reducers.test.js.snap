// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<movieAppReducer /> should return the initial state 1`] = `
Object {
  "moviesFound": 0,
  "moviesList": Array [],
  "searchInputValue": "",
  "selectedFilterOption": "title",
  "selectedMovieInfo": Object {},
  "selectedSortOption": "release",
}
`;
